samples/geotag.cpp

```diff
@@ -420,7 +420,7 @@ int timeZoneAdjust()
     struct tm local = *localtime(&now) ;
     offset          = local.tm_gmtoff ;
 
-#if DEBUG
+#if DEBUG && !defined(_MSC_VER) && !defined(__MINGW__)
     struct tm utc = *gmtime(&now);
     printf("utc  :  offset = %6d dst = %d time = %s", 0     ,utc  .tm_isdst, asctime(&utc  ));
     printf("local:  offset = %6d dst = %d time = %s", offset,local.tm_isdst, asctime(&local));
```

Comment:
The debug output in line 426 references the variable `local`. But `local` is defined in line 420 and only available on Un*x builds, not MSVC, not MinGW, not __CYGWIN__ and not OS_SOLARIS.

Reply:
But the `#if DEBUG` is already present in the `#else` branch, after checking whether `__MINGW__`, `_MSC_VER`, `__CYGWIN__` or `OS_SOLARIS` are defined, so this should be inserted only for the cases where `_MSC_VER` and `__MINGW__` are not defined.

I guess I am missing something here?

Reply:
I do not understand either what this change is suppose to fix. If we go inside that block (lines 424-427) we already know that `_MSC_VER` and `__MINGW__` are not defined, since we are in the else branch (line 419) .